WordPress/gutenberg
# Patterns: Improve the screenreader announcements for category navigation

## What?

Updated 11th of July 2023.
Closes https://github.com/WordPress/gutenberg/issues/52005

When the original issue was opened, the Site Editor patterns page did not include information about if a category was for a pattern or a template part. The original PR aimed to address that.
Now, the patterns page has visible h2 headings above the different categories. These headings are not focusable but are announced by screen readers when reading the full content of the sidebar (navigation mode).

The updated PR attempts to implement the suggestion from the comment below:

1. hide the count with aria-hidden="true"
2. add a VisuallyHidden component outside of the button that holds a descriptive text e.g.: template part, 1 item
3. reference the visually hidden description by the means of aria-describedby
4. this would require using a unique ID, probably by the means of useInstanceId.

### Testing Instructions for Screen readers
Make sure that your WordPress install has some synced block patterns.
Activate a block theme, and the screen reader.
Open the Patterns page via the WordPress admin area > Appearance > Editor > Patterns.
Use the tab key to navigate past the initial links and buttons until you have passed the "Create pattern" button.

Using VoiceOver, navigating with tab key: Announces "My patterns, button.... more content available",  Opening the more content menu announces  "1" - the number of items in the category.
Pressing the tab key again will skip past the non-focusable heading for the template parts, and announce:
"Header, button .... more content available". Opening the more content menu announces  "1" - the number of items in the category.

Using Voiceover, reading the full content of the sidebar:
Announces  "My patterns, My patterns, button, 1" followed by "Heading level 2, Template parts, list, 3 items, Header, current item, Header, button, 2".
So in this case, there is a lot of repetition.
